### PR TITLE
Adding Probes to ML Pipelines' pods

### PR DESCRIPTION
**Pull request by DharmitD**
- Adding probes to the ML pipelines' pods
- Updating delay seconds for Liveness probes
- Rearranging the probes in the same key order so the various components are congruent

### CODE REVIEW

**Comment by DharmitD** [comment]
Tested out on a local cluster by applying this kfdef: http://pastebin.test.redhat.com/1073383

Pods came up as expected, without any CrashLoopBackOff.

File: manifests/opendatahub/overlays/metadata-store-mysql/deployments/mysql.yaml [hunk 1]
```diff
@@ -31,6 +31,22 @@ spec:
...
               name: mysql
+          readinessProbe:
+            failureThreshold: 3
+            initialDelaySeconds: 5
+            periodSeconds: 10
+            successThreshold: 1
+            tcpSocket:
+              port: 3306
+            timeoutSeconds: 1
+            livenessProbe:
```
Comment (gmfrasca) [inline manifests/opendatahub/overlays/metadata-store-mysql/deployments/mysql.yaml]: This section appears to be indented incorrectly

File: manifests/opendatahub/base/deployments/metadata-envoy-deployment.yaml [hunk 2]
```diff
@@ -22,8 +22,20 @@ spec:
...
+            tcpSocket:
+              port: md-envoy
+            timeoutSeconds: 2
           ports:
             - containerPort: 9090
               name: md-envoy
             - containerPort: 9901
               name: envoy-admin
+          readinessProbe:
+            initialDelaySeconds: 3
```
Comment (gmfrasca) [inline manifests/opendatahub/base/deployments/metadata-envoy-deployment.yaml]: The values for readiness and likeness probes are exactly the same, making both probes identical.  Should readiness  have a longer threshold or duration here?

**Comment by gmfrasca** [comment]
Very minor annoyance: can we organize the probes in the same key order so the various components are congruent?  Helps with readability and reduces confusion/mistakes

File: manifests/opendatahub/base/deployments/metadata-envoy-deployment.yaml (hunk 2 above)
Comment (DharmitD) [reply manifests/opendatahub/base/deployments/metadata-envoy-deployment.yaml]: I used probes in `metadata-grpc` as a reference, and both the probes are exactly the same here:
https://github.com/opendatahub-io/ml-pipelines/blob/master/manifests/opendatahub/base/deployments/metadata-grpc-deployment.yaml#L58-L73
Sure, should we just double all thresholds for readiness in both envoy and grpc pods then?

**Comment by DharmitD** [comment]
> Very minor annoyance: can we organize the probes in the same key order so the various components are congruent? Helps with readability and reduces confusion/mistakes

Done, rearranged probes and ports to have them in the following order:
```
ports
livenessProbe
readinessProbe
```

Also, tested out the changes so far and pods come up as expected without crash looping.

File: manifests/opendatahub/base/deployments/metadata-envoy-deployment.yaml (hunk 2 above)
Comment (DharmitD) [reply manifests/opendatahub/base/deployments/metadata-envoy-deployment.yaml]: Updated `initialDelaySeconds` to 30 in the liveness probe based on the `mysql` deployment example

File: manifests/opendatahub/overlays/metadata-store-mysql/deployments/mysql.yaml (hunk 1 above)
Comment (DharmitD) [reply manifests/opendatahub/overlays/metadata-store-mysql/deployments/mysql.yaml]: Updated to have correct indentation.

**Comment by DharmitD** [comment]
Tested out latest changes on a cluster, and the pods come up and run as expected, without any crash looping.

File: manifests/opendatahub/base/deployments/metadata-writer.yaml [hunk 3]
```diff
@@ -32,4 +32,20 @@ spec:
...
+                - pidof
+                - python3
+            initialDelaySeconds: 30
+            periodSeconds: 5
+            timeoutSeconds: 2
+          readinessProbe:
+            exec:
+              command:
+                - pidof
+                - python3
```
Comment (DharmitD) [inline manifests/opendatahub/base/deployments/metadata-writer.yaml]: added check for python based on @rimolive 's suggestions, any suggestions on a more metadata writer relevant check?

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request merged**